mypy/plugins/singledispatch.py

```diff
@@ -138,24 +138,25 @@ def singledispatch_register_callback(ctx: MethodContext) -> Type:
         return register_callable
     elif isinstance(first_arg_type, CallableType):
         # TODO: do more checking for registered functions
-        register_function(ctx, ctx.type, first_arg_type)
+        return register_function(ctx, ctx.type, first_arg_type)
```

Comment:
Could we just return `first_arg_type` directly here, and `func` directly in the other case?

Reply:
There are some places in `register_function` that return early if we find an error, and in those cases `register_function` returns `ctx.default_return_type` to prevent other unexpected errors that could be misleading.

We could also have `register_function` return a `bool` that determines whether any errors were found and then use that to determine what we should return, but that would probably just make the places we call it more verbose.